inc/class-nusa-assets.php

```diff
@@ -0,0 +1,249 @@
+	 *
+	 * @var string
+	 */
+	private $critical_css = '';
+
+	/**
+	 * Using construct function to add any actions and filters
+	 */
+	public function __construct() {
+		$this->get_critical_css();
+
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_admin_assets' ] );
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_assets' ], 99 );
+		add_action( 'script_loader_tag', [ $this, 'do_script_loader_tag' ], 10, 3 );
+		add_action( 'style_loader_tag', [ $this, 'do_style_loader_tag' ], 10, 4 );
+		add_action( 'css_do_concat', [ $this, 'exclude_critical_css_concat' ], 10, 2 );
+		add_action( 'wp_head', [ $this, 'inline_critical_css' ], -1 );
+	}
+
+	/**
+	 * Singleton
+	 *
+	 * Returns a single instance of this class.
+	 */
+	public static function singleton() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Get the critical CSS as soon as possible so that other logic
+	 * can be handled without fetching it multiple times.
+	 *
+	 * @return void
+	 */
+	private function get_critical_css() {
+		$file_source           = get_template_directory() . '/assets/css/critical.min.css';
+		$critical_css_mod_time = filemtime( $file_source );
+		$mod_time_transient    = get_transient( 'critical_css_mod_time' );
+		$critical_css          = get_transient( 'critical_css' );
+
+		if ( ( empty( $critical_css ) || $critical_css_mod_time !== $mod_time_transient ) && file_exists( $file_source ) ) {
+			$critical_css = file_get_contents( $file_source );  // phpcs:ignore WordPressVIPMinimum.Performance.FetchingRemoteData.FileGetContentsUnknown
+
+			if ( ! empty( $critical_css ) ) {
+				// Strip the comments and source map.
+				$critical_css = preg_replace( '#/\\*.*?\\*/#s', '', $critical_css );
+
+				// Add full path to urls.
+				$critical_css = str_replace( 'url(../../images', 'url(' . get_template_directory_uri() . '/images', $critical_css );
+
+				set_transient( 'critical_css_mod_time', $critical_css_mod_time );
+				set_transient( 'critical_css', $critical_css );
+			}
+		}
+
+		if ( ! empty( $critical_css ) ) {
+			$this->critical_css = $critical_css;
+		}
+	}
+
+	/**
+	 * Enqueue Admin Assets
+	 *
+	 * Enqueues the necessary css and js files when the WordPress admin is loaded.
+	 */
+	public function enqueue_admin_assets() {
+		wp_enqueue_style(
+			'nusa',
+			get_stylesheet_directory_uri() . '/assets/css/wp-admin.min.css',
+			[],
+			filemtime( get_template_directory() . '/assets/css/wp-admin.min.css' )
+		);
+
+		wp_enqueue_script(
+			'nusa',
+			get_stylesheet_directory_uri() . '/assets/js/wp-admin.min.js',
+			[ 'jquery', 'media-upload' ],
+			filemtime( get_template_directory() . '/assets/js/wp-admin.min.js' ),
+			true
+		);
+	}
+
+	/**
+	 * Enqueue Assets
+	 *
+	 * Enqueues the necessary css and js files when the theme is loaded.
+	 */
+	public function enqueue_assets() {
+		$theme_path = get_template_directory();
+		$theme_uri  = get_stylesheet_directory_uri();
```

Comment:
This goes for the whole file since it is in the specified lines (94, 101, 115).
`get_template_directory*` and `get_stylesheet_directory*` functions should not be used together when enqueuing assets since they may point to different locations. `get_template_directory()` and `get_stylesheet_directory()` point to different directions.
In the case of the `_template_` functions, they will point to the theme itself no matter if in a parent or child theme.
On the other hand, for the `_stylesheet_` functions, they will point to the child theme if it is one, or itself if it is not a child theme.

For this file we should be using `get_template_directory_uri()` for all three instances that it exists.